trace_http_client.go

```diff
@@ -0,0 +1,94 @@
+package ottwirp
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	otlog "github.com/opentracing/opentracing-go/log"
+	"github.com/twitchtv/twirp"
+)
+
+// HTTPClient as an interface that models *http.Client.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// TraceHTTPClient wraps a provided http.Client and tracer for instrumenting
+// requests.
+type TraceHTTPClient struct {
+	client HTTPClient
+	tracer opentracing.Tracer
+}
+
+func NewTraceHTTPClient(client HTTPClient, tracer opentracing.Tracer) HTTPClient {
```

Comment:
i would go the "accept interfaces, return structs" maxim here. if you're exporting the `TraceHTTPClient`, no reason not to return it (since it implements the interface, any function that takes it will still work and you can enforce this with `var _ HTTPClient = (*TraceHTTPClient)(nil)`)

some additional info on this:

https://blog.chewxy.com/2018/03/18/golang-interfaces/
https://mycodesmells.com/post/accept-interfaces-return-struct-in-go
(this doesn't apply because you use the interface where you have the implementation, but good to see anyway) https://twitter.com/davecheney/status/942593128355192832?lang=en